### ISSUE TITLE
Adding a new article to a published issue does not specifically set sequence

### ISSUE THREAD

**Issue by jnugent** [pkp/pkp-lib#8998]
**Describe the bug**
When a submission is first created, the publication record associated with it gets a `seq` value of 0, which makes sense since it isn't in an issue yet.  When the submission is ultimately assigned to an issue and the issue is published, these 0 `seq` values are not re-sequenced into incrementing integers.  This results in the articles within a section randomly ordering the articles and this order will change from page load to page load.  The only way around this is to specify a sort order within the issue's table of contents.  That's probably fine.  However, when new articles are added to an issue with an already set sort order, the 0 `seq` values cause these articles to appear at the top of the section, and they will flip randomly around each time the page is loaded.

The Quick Submit plugin will increment the `seq` value on submissions as it builds an issue so perhaps there is some code here that can be reused.  At the least, maybe the query builder for fetching submissions can also sort according to `submission_id` so at least there is a unique value in case the sequence values are the same.

Our clients who use this workaround for continuous publication have also asked us to examine sorting by the `date_published` field within the publication object, but since this field does not contain minutes and seconds it results in the same behaviour when more than one article is published on the same day.

**To Reproduce**
1.  Create several submissions
2. Add these submissions to an issue without sorting the table of contents
3. publish the issue, and reload the TOC.  
4. Specifically sort the issue, the TOC is now locked.
5. Add several more articles to the issue without sorting them.
6. Load the TOC, they should flip around.

**What application are you using?**
OJS 3.3.0.14

**Additional information**
Issue TOC ordering and sorting is related to this other issue:  https://github.com/pkp/pkp-lib/issues/8564

**Comment by ewhanson** [comment]
@jnugent, is this issue still present when you stop after step 3 above, i.e. when you keep the unsorted table of contents but do not add any more items?

**Comment by ewhanson** [comment]
@jnugent, thanks for the details. This came up with some of the Coalition Publica work as well so I'm going to dive into it. I haven't done any testing yet, but do you know if this is also a problem in 3.4 and later? (I'm assuming if it wasn't fixed in 3.3, it's still present, but want to know what I'm getting myself into).

**Comment by jnugent** [comment]
Hi @ewhanson I just tested this in our 3.4 testdrive.  If I schedule a submission for publication in an empty issue, it gets a `seq` of 0. If add a second article to that same issue, it gets a `seq` of 1.  So this might not be a problem in 3.4.  I think @jonasraoni may have worked on this.

**Comment by RewindLife** [comment]
Hi @jnugent and @jonasraoni. Is there any update on this bug fix? Will the fix be released toether with 3.4 update? Thanks.

**Comment by ewhanson** [comment]
@aupress I'm currently investigating the status of this. Will provide an update here once a fix has been made.

@jnugent, I'm testing it locally on `stable-3_4_0` and still seeing all publications assigned to a `seq` of 0. Do you know at which point in the publication process the second article was getting a `seq` of 1?

**Comment by ewhanson** [comment]
Thanks @jnugent, that's helpful (and glad I'm not going crazy). I'm working on a fix to set the `seq` when it's assigned to an issue and will see how that behaves.

**Comment by aupress** [comment]
@ewhanson @jnugent 
Could you please confirm that this issue was fixed in the most recent OJS update (OJS 3.4.0-7 was released on August 23, 2024)? 
Our journal is on OJS 3.3.0.14 and we are still experiencing the TOC reshuffle issue as of November 2024. Thank you!

**Comment by Tribunal33** [comment]
I'm not sure when this was fixed, but I couldn't reproduce the behavior on version 3.5.  The order seems to be respected (I didn't look into DB structure) but I couldn't get it to "flip" when reloading the TOC.  Not sure of the expected behavior if you order a few of the articles assigned to the issue and then add more to it afterwards.  Seems to maintain an order which puts the last published article at the above previously sorted ones.

<img width="1809" height="992" alt="Image" src="https://github.com/user-attachments/assets/93a6f0e9-ac9b-49bb-9000-e6a21acde7fa" />

**Comment by bozana** [comment]
a note coming from this issue: https://github.com/pkp/pkp-lib/issues/11752:
When a publication changes section it keeps the seq number, i.e. the re-ordering is not considered in such a case. It can be fixed by manually ordering it in the TOC grid. I am not sure if we want to change something here... but just to mention it...

**Comment by Tribunal33** [comment]
@bozana did #11752 fix the flipping issue seen with a race condition?  If so I think we can close this one?

**Comment by bozana** [comment]
@Tribunal33, this is a separate issue. I have just improved code for custom section order. Publication order, in all possible scenarios, probably still needs to be checked/fixed.
